gtsfm/frontend/detector_descriptor/detector_descriptor_base.py

```diff
@@ -13,16 +13,20 @@
 import gtsfm.utils.logger as logger_utils
 from gtsfm.common.image import Image
 from gtsfm.common.keypoints import Keypoints
+from gtsfm.ui.registry import GTSFMProcess, UiMetadata
 
 logger = logger_utils.get_logger()
 
 
-class DetectorDescriptorBase(metaclass=abc.ABCMeta):
+class DetectorDescriptorBase(GTSFMProcess):
     """Base class for all methods which provide a joint detector-descriptor to work on an image.
 
     This class serves as a combination of individual detector and descriptor.
     """
 
+    def get_ui_metadata() -> UiMetadata:
```

Comment:
Let's add a 1-line docstring to all the `get_ui_metadata()` function defintions.